From 2aeb12709d2ff17470d653523b56040f56fced4c Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Mon, 18 Apr 2022 18:26:31 +0000
Subject: [PATCH] libstdc++: fix DCHECK_EQ/NE in
 content::FirstPartySetsDatabase

There is no CheckOpValueStr() for std::unique_ptr. Use get() in
DCHECK_EQ to retrieve the pointer value. Change DCHECK_NE to
DCHECK.

Bug: 957519
Change-Id: I57440d1863f05480dd4775cf314a98caec542b5a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3585456
Reviewed-by: Alex Moshchuk <alexmos@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#993372}
---
 .../database/first_party_sets_database.cc                 | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/content/browser/first_party_sets/database/first_party_sets_database.cc b/content/browser/first_party_sets/database/first_party_sets_database.cc
index 54e36e4a87b80..18a90347fb6cd 100644
--- a/content/browser/first_party_sets/database/first_party_sets_database.cc
+++ b/content/browser/first_party_sets/database/first_party_sets_database.cc
@@ -173,7 +173,7 @@ bool FirstPartySetsDatabase::LazyInit() {
   if (db_status_ != InitStatus::kUnattempted)
     return db_status_ == InitStatus::kSuccess;
 
-  DCHECK_EQ(db_, nullptr);
+  DCHECK_EQ(db_.get(), nullptr);
   db_ = std::make_unique<sql::Database>(sql::DatabaseOptions{
       .exclusive_locking = true, .page_size = 4096, .cache_size = 32});
   db_->set_histogram_tag("FirstPartySets");
@@ -196,7 +196,7 @@ bool FirstPartySetsDatabase::LazyInit() {
 }
 
 bool FirstPartySetsDatabase::OpenDatabase() {
-  DCHECK_NE(db_, nullptr);
+  DCHECK(db_);
   if (db_->is_open() || db_->Open(db_path_)) {
     db_->Preload();
     return true;
@@ -206,7 +206,7 @@ bool FirstPartySetsDatabase::OpenDatabase() {
 
 void FirstPartySetsDatabase::DatabaseErrorCallback(int extended_error,
                                                    sql::Statement* stmt) {
-  DCHECK_NE(db_, nullptr);
+  DCHECK(db_);
   // Attempt to recover a corrupt database.
   if (sql::Recovery::ShouldRecover(extended_error)) {
     // Prevent reentrant calls.
@@ -329,4 +329,4 @@ bool FirstPartySetsDatabase::Destroy() {
   return base::DeleteFile(db_path_);
 }
 
-}  // namespace content
\ No newline at end of file
+}  // namespace content
